### PR TITLE
distributed-builds.md: Clarify warning ssh access requirements

### PR DESCRIPTION
**Pull request by doronbehar**
# Motivation

Makes a bit clearer a misunderstanding I've had when I read that warning.

# Context

Somewhat related: https://github.com/NixOS/nix/issues/7380 .

# Checklist for maintainers

Maintainers: tick if completed or explain if not relevant

 - [ ] agreed on idea
 - [ ] agreed on implementation strategy
 - [ ] tests, as appropriate
   - functional tests - `tests/**.sh`
   - unit tests - `src/*/tests`
   - integration tests - `tests/nixos/*`
 - [ ] documentation in the manual
 - [ ] documentation in the internal API docs
 - [ ] code and comments are self-explanatory
 - [ ] commit message explains why the change was made
 - [ ] new feature or incompatible change: updated release notes

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: doc/manual/src/advanced-topics/distributed-builds.md [hunk 1]
```diff
@@ -38,11 +38,11 @@ contains Nix.
 
 > **Warning**
 >
-> If you are building via the Nix daemon, it is the Nix daemon user
-> account (that is, `root`) that should have SSH access to the remote
-> machine. If you can’t or don’t want to configure `root` to be able to
-> access to remote machine, you can use a private Nix store instead by
-> passing e.g. `--store ~/my-nix`.
+> If you are building via the Nix daemon, it is the Nix daemon user account
+> (that is, `root`) that should have SSH access to a certain user on the remote
```
Comment (fricklerhandwerk) [inline doc/manual/src/advanced-topics/distributed-builds.md:42]:
```suggestion
> (that is, `root`) that should have SSH access to a user on the remote
```

If you say "certain", I understand it may be a particular one. If that doesn't matter, keep it "a user". If we can provide more information (such as a trusted user), add that (and add links to relevant documentation, such as the trusted users setting).

Also, please let's use one sentence per line in markdown, because reflows produce diff noise that is hard to parse and review.

**Review by fricklerhandwerk — APPROVED** [review]

File: doc/manual/src/advanced-topics/distributed-builds.md (hunk 1 above)
Comment (doronbehar) [reply doc/manual/src/advanced-topics/distributed-builds.md]: > If you say "certain", I understand it may be a particular one.

You are correct.

> If that doesn't matter, keep it "a user".

I only wanted to emphasize that it doesn't need to be necessarily root on the remote machine.

> If we can provide more information (such as a trusted user), add that (and add links to relevant documentation, such as the trusted users setting).

According to my experience the remote user doesn't need to be trusted.

> Also, please let's use one sentence per line in markdown (even if it's not consistent everywhere yet), because reflows produce diff noise that is hard to parse and review.

I won't mind - pushed a fix now. The diff noise is reduced, but the line breaks look a bit random now.

**Comment by fricklerhandwerk** [comment]
I took the liberty to force push the branch so the rewrap is in a single commit.
